src/renderer/components/kube-object-menu/kube-object-menu.tsx

```diff
@@ -41,11 +41,11 @@ class NonInjectedKubeObjectMenu<TKubeObject extends KubeObject> extends React.Co
   }
 
   get isEditable() {
-    return this.props.editable ?? Boolean(this.store?.patch);
+    return this.props.editable ?? this.store?.showEditButton ?? false;
```

Comment:
Having `showEditButton` and other field on `KubeObjectStore` just weird as heck.
Why only `props.editable` not enough to control this items?

Reply:
I agree it is extremely weird.

Reply:
It is not enough because the problem here is the fact that the extension API does not have a mechanism to remove these buttons for any kube objects, let alone for new kube objects which an extension might actually control.

Another solution would be to require extension developers to add these buttons explicitly but that is a breaking change.